src/os/unix/wayland.rs

```diff
@@ -0,0 +1,1182 @@
+        surface.damage(0, 0, i32::max_value(), i32::max_value());
+        surface.commit();
+
+        let xdg_config = Rc::new(RefCell::new(None));
+        {
+            let xdg_config = xdg_config.clone();
+            xdg_surface.quick_assign(move |_xdg_surface, event, _| {
+                use wayland_protocols::xdg_shell::client::xdg_surface::Event;
+
+                //Acknowledge only the last configure
+                if let Event::Configure { serial } = event {
+                    *xdg_config.borrow_mut() = Some(serial);
+                }
+            });
+        }
+
+        let cursor = wayland_cursor::load_theme(None, 16, &shm);
+        let cursor_surface = comp.create_surface();
+
+        Ok((
+            Self {
+                _display: display,
+                wl_display,
+                _comp: comp,
+                _base: xdg_wm_base,
+                surface,
+                xdg_surface,
+                toplevel: xdg_toplevel,
+                _shm: shm,
+                event_queue: event_q,
+                _seat: seat,
+                xdg_config,
+                cursor,
+                cursor_surface,
+                buf_pool,
+            },
+            input,
+        ))
+    }
+
+    fn set_geometry(&self, pos: (i32, i32), size: (i32, i32)) {
+        self.xdg_surface
+            .set_window_geometry(pos.0, pos.1, size.0, size.1);
+    }
+
+    fn set_title(&self, title: &str) {
+        self.toplevel.set_title(title.to_owned());
+    }
+
+    fn set_no_resize(&self, size: (i32, i32)) {
+        self.toplevel.set_max_size(size.0, size.1);
+        self.toplevel.set_min_size(size.0, size.1);
+    }
+
+    //Sets a specific cursor style
+    fn update_cursor(&mut self, cursor: &str) {
+        let csr = self.cursor.get_cursor(cursor).unwrap();
+        let img = csr.frame_buffer(0).unwrap();
+        self.cursor_surface.attach(Some(&*img), 0, 0);
+        self.cursor_surface.damage(0, 0, 32, 32);
+        self.cursor_surface.commit();
+    }
+
+    //resizes when buffer is bigger or less
+    fn update_framebuffer(&mut self, buffer: &[u32], size: (i32, i32)) {
+        use std::io::{Seek, SeekFrom};
+
+        let (mut fd, buf) = self.buf_pool.get_buffer(size);
+
+        fd.seek(SeekFrom::Start(0)).unwrap();
+        let slice = unsafe {
+            std::slice::from_raw_parts(
+                buffer[..].as_ptr() as *const u8,
+                buffer.len() * std::mem::size_of::<u32>(),
+            )
+        };
+        fd.write_all(&slice[..]).unwrap();
+        fd.flush().unwrap();
+
+        //Acknowledge the last configure event
+        if let Some(serial) = (*self.xdg_config.borrow_mut()).take() {
+            self.xdg_surface.ack_configure(serial);
+        }
+
+        self.surface.attach(Some(buf), 0, 0);
+        self.surface
+            .damage(0, 0, i32::max_value(), i32::max_value());
+        self.surface.commit();
+    }
+
+    //Keyboard, Pointer, Touch
+    fn _check_capabilities(
```

Comment:
Considering this appears unused, what is its purpose in the first place?

Reply:
Firstly I thought it is mandatory to check the capabilities, but then I left it out there. I currently consider removing it.